arch/Kconfig

```diff
@@ -445,6 +445,7 @@ config MPU_REQUIRES_NON_OVERLAPPING_REGIONS
 menuconfig FLOAT
 	bool "Floating point"
 	depends on CPU_HAS_FPU
+	depends on ARM || X86
```

Comment:
why this? isn't CPU_HAS_FPU enough?

Reply:
In fact CPU_HAS_FPU is selected right now in the two configs exposed by soc/arc/snps_emsk, so this looks to me like this patch is going to break floating point on ARC nsim targets (no idea if it worked, mind you, but it's going to be disabled unexpectedly by this patch).

Reply:
Responded below, @andrewboie @andyross